tar.go

```diff
@@ -29,11 +30,13 @@ func NewTarStreamer(r io.Reader, keywords []string) Streamer {
 		tarReader:  tar.NewReader(pR),
 		keywords:   keywords,
 	}
-	go ts.readHeaders() // I don't like this
+
+	go ts.readHeaders()
```

Comment:
@cyphar it is intended to be a pipe-like operation. Such that as the caller is fetching a remote tar archive, that the manifest is built as the stream is read. By the time the archive is local, but not yet extracted, the manifest is available.
Otherwise, a `Walk()` requires an additional pass through the archive.